src/Service/Taf/NoaaService.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Service\Taf;
+
+use App\Traits\ZMQContextTrait;
+use Cake\Console\ConsoleIo;
+use Cake\Datasource\ModelAwareTrait;
+use Cake\Http\Client;
+
+/**
+ * @property \App\Model\Table\AirportsTable $Airports
+ * @property \App\Model\Table\TafTable $Taf
+ */
+class NoaaService
+{
+    use ModelAwareTrait;
+    use ZMQContextTrait;
+
+    /**
+     * @var string
+     */
+    protected const _NOAA_URL = 'https://aviationweather.gov/adds/dataserver_current/httpparam';
+
+    /**
+     * @var \Cake\Console\ConsoleIo|null
+     */
+    protected ?ConsoleIo $_io;
+
+    /**
+     * @var \Cake\Http\Client
+     */
+    protected Client $_client;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->loadModel('Airports');
+        $this->loadModel('Taf');
+
+        $this->_client = new Client();
+    }
+
+    /**
+     * @FIXME split airports into more than one request
+     * @return void
+     */
+    public function getTaf(): void
+    {
+        $airports = $this->Airports->find()
+            ->select(['icao'])
+            ->enableHydration(false)
+            ->all()
+            ->extract('icao')
+            ->toArray();
+
+        if($airports) {
+            $rawTaf = $this->_fetchTaf($airports);
+    
+            if($rawTaf && $rawTaf['data']['@attributes']['num_results'] && $rawTaf['data']['TAF']) {
```

Comment:
We should check if the array actually contains values. Something like `!empty()` or `isset()` depending on how the API returns it's data.